docs: Update Kubernetes tutorial to create SAN certificate

nsilve opened the pull request:
## Short description

Create SAN certificate for OPA service because Go 1.15 deprecated CN support.

## Long description

Go 1.15 [deprecated CN support](https://golang.org/doc/go1.15#commonname) so latest Kubernetes `1.19` which is built on that version

```shell
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.2", GitCommit:"f5743093fd1c663cb0cbc89748f730662345d44d", GitTreeState:"clean", BuildDate:"2020-09-16T13:41:02Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.2", GitCommit:"f5743093fd1c663cb0cbc89748f730662345d44d", GitTreeState:"clean", BuildDate:"2020-09-16T13:32:58Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}
```
has issues with CN only usage.

K8s API server cannot communicate with OPA and it logs the following:
```
W1010 17:08:38.525599       1 dispatcher.go:129] Failed calling webhook, failing open validating-webhook.openpolicyagent.org: failed calling webhook "validating-webhook.openpolicyagent.org": Post "https://opa.opa.svc:443/?timeout=30s": x509: certificate relies on legacy Common Name field, use SANs or temporarily enable Common Name matching with GODEBUG=x509ignoreCN=0
E1010 17:08:38.525631       1 dispatcher.go:130] failed calling webhook "validating-webhook.openpolicyagent.org": Post "https://opa.opa.svc:443/?timeout=30s": x509: certificate relies on legacy Common Name field, use SANs or temporarily enable Common Name matching with GODEBUG=x509ignoreCN=0
```

and OPA logs the following:
```
2020/10/10 17:08:38 http: TLS handshake error from 10.157.171.192:14074: remote error: tls: bad certificate
```

Kubernetes tutorial updated to generate SAN certificate using the same hostname as in CN field.

---

Signed-off-by: Nikos Silvestros <nsilvestros@gmail.com>

patrick-east commented:
Testing this out it doesn't look like the default openssl version on OSX supports the `-addtext` flag:

```
{12:12} ~ ❯ openssl req -new -key server.key -out server.csr -subj "/CN=opa.opa.svc" -addext "subjectAltName = DNS:opa.opa.svc" -config server.conf

unknown option -addext
<snip>
```

I think we can make it work for more environments by setting the SAN in the `server.conf` we create in the step right before this. Adding a few lines to the end of it like:

```
...
[ v3_req ]
...
subjectAltName = @alt_names
[alt_names]
DNS.1 = opa.opa.svc
```

nsilve commented:
> Testing this out it doesn't look like the default openssl version on OSX supports the `-addtext` flag:
> 
> ```
> {12:12} ~ ❯ openssl req -new -key server.key -out server.csr -subj "/CN=opa.opa.svc" -addext "subjectAltName = DNS:opa.opa.svc" -config server.conf
> 
> unknown option -addext
> <snip>
> ```
> 
> I think we can make it work for more environments by setting the SAN in the `server.conf` we create in the step right before this. Adding a few lines to the end of it like:
> 
> ```
> ...
> [ v3_req ]
> ...
> subjectAltName = @alt_names
> [alt_names]
> DNS.1 = opa.opa.svc
> ```

Updated based on last comment.
I added all OPA service hostname references into OpenSSL configuration file and removed them from command line to gather all of them in the same file/documentation section.

patrick-east reviewed (approved):
LGTM
